http/transport-component.md

```diff
@@ -0,0 +1,111 @@
+# HTTP Transport Component <!-- omit in toc -->
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+| --------------- | ------------- | ------ | --------------- |
+| 1A              | Working Draft | Active | r0, 2023-05-31  |
+
+Authors: [@marcopolo]
+
+Interest Group: [@marcopolo], [@mxinden], [@marten-seemann]
+
+[@marcopolo]: https://github.com/marcopolo
+[@mxinden]: https://github.com/mxinden
+[@marten-seemann]: https://github.com/marten-seemann
+
+## Table of Contents <!-- omit in toc -->
+- [Context](#context)
+- [What is an HTTP transport](#what-is-an-http-transport)
+- [Multiaddr representation](#multiaddr-representation)
+- [HTTP Paths](#http-paths)
+
+
+## Context
+
+This document is only about advertising support for an HTTP transport. It
+doesn't make any assertions about how libp2p should interact with that
+transport. That will be defined in a future document.
+
+This exists to clarify the role of the `/http` component in Multiaddrs early to
+avoid confusion and conflicting interpretations.
+
+## What is an HTTP transport
+
+An HTTP transport is simply a node that can speak some standardized version of
+HTTP. Intuitively if you can `curl` it with HTTP, then it speaks HTTP.
+
+Most environments will have a way to create an HTTP Client and Server, and the
+specific HTTP version used will be opaque. We use the `/http` component at the
+end of the multidadr to signal that this server supports an HTTP transport. The
+end user agent decides on HTTP version to use, based on the multiaddr prefix,
+application, server negotiation, and specific use case. This follows what
+existing `http://` URL implementations do.
+
+## Multiaddr representation
+
+The multiaddr of a node with an HTTP transport ends with `/http` and is prefixed
+by information that would let an HTTP client know how to reach the server
+(remember that multiaddrs are [interpreted right to
+left](https://github.com/multiformats/multiaddr#interpreting-multiaddrs)). 
+
+The following are examples of multiaddrs for HTTP transport capable nodes:
+
+* `/dns/example.com/tls/http`
+* `/ip4/1.2.3.4/tcp/443/tls/http`
+* `/ip6/2001:0db8:85a3:0000:0000:8a2e:0370:7334/tcp/443/tls/http`
+* `/ip4/1.2.3.4/udp/50781/tls/http`
+
+
+## HTTP Paths
```

Comment:
I almost want to remove this section since it's kind of a tangent. It's useful in making the case why `/httppath` shouldn't be appended to `/http` and what you can do instead, but it also muddies the spec.

Reply:
Hm.. maybe replace the tangent with a list of HTTP Semantics that one would be tempted to expose after `/http`, like I've hinted in https://github.com/ipni/go-libipni/issues/42#issuecomment-1570503551?

> After all, paths are not special, it is only one of many attributes that may impact HTTP endpoint's behavior.  If we allow HTTP paths, the unanswered question is: how one adds basic auth or bearer token headers? or a cookie? or an Accept header for content type version negotiation? [Do we add `/foo` for every HTTP feature? Where is the line in the sand?]
> 
> There are prior discussions in
> - https://github.com/multiformats/multiaddr/issues/63
> - https://github.com/multiformats/multiaddr/issues/87

Reply:
Should this section make recommendations on how paths should be handled? The HTTP rest APIs commonly support this. I think at the very least there should be recommendations on how should one handle HTTP API path prefixes when expressed as multiaddr.

Reply:
Many thanks for adding this @MarcoPolo 

Using private ranges to specify path is certainly one way to do it. However, I am conscious of the increase in barrier for adoption. This is something that hopefully we can iterate on with the help of the lip2p community as HTTP over libp2p matures.